tests/unit/Entity/ItemIdSetTest.php

```diff
@@ -16,6 +16,35 @@
  */
 class ItemIdSetTest extends \PHPUnit_Framework_TestCase {
 
+	public function testGetIterator() {
+		$set = new ItemIdSet();
+		$iterator = $set->getIterator();
+
+		$this->assertInstanceOf( 'Traversable', $iterator );
+	}
+
+	/**
+	 * @dataProvider serializationsProvider
+	 */
+	public function testGetSerializations( array $expected ) {
```

Comment:
I also tend to make tests too complex in order to be DRY. As @JeroenDeDauw wrote, this introduces the risk of reproducing mistakes you did in the production code, thereby masking these mistakes in the production code.

What I would definitely recommend is moving the instantiation to the provider. The test method should not have to prepare its test input, that should be the responsibility of the provider in my opinion.